scalecube-configuration/src/main/java/io/scalecube/configuration/ConfigurationServiceImpl.java

```diff
@@ -1,133 +1,100 @@
-        .then(Mono.fromCallable(ServiceOperationFactory::fetchAll))
-        .flatMapMany(
-            operation -> Flux.from(operation.execute(request, context(OperationType.List))))
-        .subscribeOn(Schedulers.parallel())
-        .doOnComplete(() -> logger.debug("entries: exit: request: {}", request))
-        .doOnError(th -> logger.error("entries: request: {}, error: {}", request, th));
+    verifyRequest(request);
+    return accessControl
+        .check(request.token().toString(), ConfigurationService.CONFIG_ENTRIES)
+        .map(p -> this.repository.fetchAll(p.tenant(), request.repository()))
+        .flatMapMany(fluxDoc -> fluxDoc.map(doc -> new FetchResponse(doc.key(), doc.value())));
   }
 
   @Override
   public Mono<Acknowledgment> save(SaveRequest request) {
-    return Mono.fromRunnable(() -> logger.debug("save: enter: request: {}", request))
-        .then(Mono.fromCallable(ServiceOperationFactory::saveEntry))
-        .flatMap(operation -> Mono.from(operation.execute(request, context(OperationType.Write))))
-        .subscribeOn(Schedulers.parallel())
-        .doOnSuccess(result -> logger.debug("save: exit: request: {}", request))
-        .doOnError(th -> logger.error("save: request: {}, error: {}", request, th));
+    verifyRequest(request);
+    return accessControl
+        .check(request.token().toString(), ConfigurationService.CONFIG_SAVE)
+        .flatMap(
+            p ->
+                this.repository.save(
+                    p.tenant(),
+                    request.repository(),
+                    Document.builder()
+                        .id(IdGenerator.generateId())
+                        .key(request.key())
+                        .value(request.value())
+                        .build()))
+        .thenReturn(ACK);
   }
 
   @Override
   public Mono<Acknowledgment> delete(DeleteRequest request) {
-    return Mono.fromRunnable(() -> logger.debug("delete: enter: request: {}", request))
-        .then(Mono.fromCallable(ServiceOperationFactory::deleteEntry))
-        .flatMap(operation -> Mono.from(operation.execute(request, context(OperationType.Delete))))
-        .subscribeOn(Schedulers.parallel())
-        .doOnSuccess(result -> logger.debug("delete: exit: request: {}", request))
-        .doOnError(th -> logger.error("delete: request: {}, error: {}", request, th));
-  }
-
-  private ServiceOperationContext context(OperationType operationType) {
-    return ServiceOperationContext.builder()
-        .operationType(authorizationService)
-        .operationType(operationType)
-        .dataAccess(dataAccess)
-        .tokenVerifier(tokenVerifier)
-        .build();
-  }
-
-  public static Builder builder() {
-    return new Builder();
+    verifyRequest(request);
+    return accessControl
+        .check(request.token().toString(), ConfigurationService.CONFIG_DELETE)
+        .flatMap(p -> this.repository.delete(p.tenant(), request.repository(), request.key()))
+        .thenReturn(ACK);
   }
 
```

Comment:
General comment regqrding the Ack responses - isn't it better to replace it with smth more meaningful? like, repo name in create repository req, saved entity in case of save etc